src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -233,6 +235,22 @@ public void toggleFieldCentric() {
 		m_fieldCentric = !m_fieldCentric;
 	}
 
+    // toggle precision mode for driving
+	public void togglePrecisionMode() {
+		m_precisionMode = !m_precisionMode;
+		m_maxVelocity = m_precisionMode ? MAX_VELOCITY_PRECISION_MODE : MAX_VELOCITY_METERS_PER_SECOND;
+		m_maxAngularVelocity = m_precisionMode ? MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND_PRECISION_MODE
+				: MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND;
+	}
+
+    // lock wheels in x position to resist pushing
+	public void lockWheels() {
+        m_swerveModules[0].set(0.0, Rotation2d.fromDegrees(45).getRadians());
+        m_swerveModules[1].set(0.0, Rotation2d.fromDegrees(-45).getRadians());
+        m_swerveModules[2].set(0.0, Rotation2d.fromDegrees(-45).getRadians());
+        m_swerveModules[3].set(0.0, Rotation2d.fromDegrees(45).getRadians());
+	}
```

Comment:
This is a little silly. You are creating a Rotation2d from degrees and then converting it to radians and then throwing it away. Repeat 3 more times, with almost the same value.
Create a double to hold the radian value of 45 degrees. Use "Math.toRadians()". Then just reuse that value, with appropriate minus signs where needed.